pkg/validation/limits.go

```diff
@@ -478,6 +480,18 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 		false,
 		"Enable persistence of patterns detected at ingest. When enabled, patterns for pushed streams will be written back into Loki as a special __pattern__ stream.",
 	)
+	f.DurationVar(
+		(*time.Duration)(&l.PatternPersistenceGranularity),
+		"limits.pattern-persistence-granularity",
+		0,
+		"The time granularity for persisting patterns. Controls how many data points are written when patterns are flushed. Set to 0 to use the default from the pattern ingester configuration.",
+	)
+	f.Float64Var(
+		&l.PatternRateThreshold,
+		"limits.pattern-rate-threshold",
+		1.0,
+		"Minimum pattern rate (samples per second) required for a pattern to be persisted. Patterns with lower rates will be filtered out during persistence. Default: 1.0",
```

Comment:
The default value is added to the `-h` output and when the docs are generated.

```suggestion
		"Minimum pattern rate (samples per second) required for a pattern to be persisted. Patterns with lower rates will be filtered out during persistence.",
```